src/main.cpp

```diff
@@ -2187,450 +2158,69 @@ int main(int arg_count, char const **arg_ptr) {
-			isize total_file_size = 0;
-			for_array(i, p->packages) {
-				files += p->packages[i]->files.count;
-				for_array(j, p->packages[i]->files) {
-					AstFile *file = p->packages[i]->files[j];
-					total_file_size += file->tokenizer.end - file->tokenizer.start;
-				}
-			}
-			gb_printf_err("total_file_size:                %lld B\n", total_file_size);
-			gb_printf_err("lines:                          %lld\n", lines);
-			gb_printf_err("files:                          %lld\n", files);
-			gb_printf_err("tokens:                         %lld\n", tokens);
-			gb_printf_err("packages:                       %lld\n", packages);
-		}
-	#endif
+	timings_start_section(timings, str_lit("LLVM API Code Gen"));
+	lbGenerator gen = {};
+	if (!lb_init_generator(&gen, &checker)) {
+		return 1;
+	}
+	lb_generate_code(&gen);
 
-		if (run_output) {
-		#if defined(GB_SYSTEM_WINDOWS)
-			return system_exec_command_line_app("odin run", "%.*s.exe %.*s", LIT(gen.output_base), LIT(run_args_string));
-		#else
-			//NOTE(thebirk): This whole thing is a little leaky
-			String output_ext = {};
-			String complete_path = concatenate_strings(heap_allocator(), gen.output_base, output_ext);
-			complete_path = path_to_full_path(heap_allocator(), complete_path);
-			return system_exec_command_line_app("odin run", "\"%.*s\" %.*s", LIT(complete_path), LIT(run_args_string));
-		#endif
-		}
+	temp_allocator_free_all(&temporary_allocator_data);
 
-		return 0;
-#else
-		gb_printf_err("LLVM C API backend is not supported on this platform yet\n");
-		return 1;
-#endif
-	} else {
-		irGen ir_gen = {0};
-		if (!ir_gen_init(&ir_gen, &checker)) {
+	switch (build_context.build_mode) {
+	case BuildMode_Executable:
+	case BuildMode_DynamicLibrary:
+		i32 result = linker_stage(&gen);
+		if (result != 0) {
 			return 1;
 		}
-		// defer (ir_gen_destroy(&ir_gen));
-
-
-		timings_start_section(timings, str_lit("llvm ir gen"));
-		ir_gen_tree(&ir_gen);
-
-		temp_allocator_free_all(&temporary_allocator_data);
-
-		timings_start_section(timings, str_lit("llvm ir opt tree"));
-		ir_opt_tree(&ir_gen);
-
-		temp_allocator_free_all(&temporary_allocator_data);
-
-		timings_start_section(timings, str_lit("llvm ir print"));
-		print_llvm_ir(&ir_gen);
-
-		temp_allocator_free_all(&temporary_allocator_data);
-
-
-		String output_name = ir_gen.output_name;
-		String output_base = ir_gen.output_base;
-
-		build_context.optimization_level = gb_clamp(build_context.optimization_level, 0, 3);
-
-		timings_start_section(timings, str_lit("llvm-opt"));
-		exec_llvm_opt(output_base);
+		break;
+	}
 
-		timings_start_section(timings, str_lit("llvm-llc"));
-		exec_llvm_llc(output_base);
+	if (build_context.show_timings) {
+		show_timings(&checker, timings);
+	}
 
-		if (build_context.build_mode == BuildMode_Object) {
-			// Ignore the linker
-			if (build_context.show_timings) {
-				show_timings(&checker, timings);
+	remove_temp_files(gen.output_base);
+
+#if defined(GB_COMPILER_MSVC)
+	if (false) {
```

Comment:
`if(false)` Why not #ifdef it out completely?

Reply:
Oversight